airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java

```diff
@@ -549,6 +566,30 @@ private static void validateSchema(final RecordSchemaValidator recordSchemaValid
     }
   }
 
+  private static void populatedStreamToSelectedFields(ConfiguredAirbyteCatalog catalog,
+                                                      Map<AirbyteStreamNameNamespacePair, List<String>> streamToSelectedFields) {
+    for (final var s : catalog.getStreams()) {
+      final List<String> selectedFields = new ArrayList<>();
+      final JsonNode propertiesNode = s.getStream().getJsonSchema().findPath("properties");
+      if (propertiesNode.isObject()) {
+        propertiesNode.fieldNames().forEachRemaining((fieldName) -> selectedFields.add(fieldName));
+      } else {
+        throw new RuntimeException("No properties node in stream schema");
+      }
+      streamToSelectedFields.put(AirbyteStreamNameNamespacePair.fromConfiguredAirbyteSteam(s), selectedFields);
+    }
+  }
+
+  private static void filterSelectedFields(Map<AirbyteStreamNameNamespacePair, List<String>> streamToSelectedFields, AirbyteMessage airbyteMessage) {
+    final AirbyteRecordMessage record = airbyteMessage.getRecord();
```

Comment:
Similar to what happens in the private method `validateSchema` in this class, probably worth an early return if `message.getRecord() == null`. In fact I wonder if this current implementation would break if there were `airbyteMessages` that weren't `records` but something else like `state`?